### PR TITLE
[#6858]Platform: Preflight checks for NFS backup storage configuration

### PR DESCRIPTION
**Pull request by jitendra-12113**
**Heading:** 
[#6858]Platform: Preflight checks for NFS backup storage configuration
**Description:**
When user initiates a manual backup -or- a scheduled backup, check whether
NFS path is a valid path on each of the database nodes before starting the backup task .
Fail the task right away before creating the snapshots for the backup.
**Testing:**
Configured the NFS backup storage PATH and created a AWS cloud provider
Then using cloud provider, created a multinode/single node universe
Ran sample Apps, which created table for backup.
Then after tried to create backup.
It will fail immediately if the storage path is not mounted on anyone of
the node.

Added unit Tests as well. It's working fine.

**Test Results:**

[info] Test run started
[info] Test com.yugabyte.yw.common.TableManagerTest.testCreateS3Backup started
[info] Test com.yugabyte.yw.common.TableManagerTest.testCreateGcsBackup started
[info] Test com.yugabyte.yw.common.TableManagerTest.testBulkImportWithSpecificInstanceCount started
[info] Test com.yugabyte.yw.common.TableManagerTest.testRestoreGcsBackup started
[info] Test com.yugabyte.yw.common.TableManagerTest.testBulkImportWithDefaultInstanceCount started
[info] Test com.yugabyte.yw.common.TableManagerTest.testCreateBackupWithSSHUser started
[info] Test com.yugabyte.yw.common.TableManagerTest.testCreateNfsBackup started
[info] Test com.yugabyte.yw.common.TableManagerTest.testCreateBackupKubernetesWithTLS started
[info] Test com.yugabyte.yw.common.TableManagerTest.testRestoreNfsBackup started
[info] Test com.yugabyte.yw.common.TableManagerTest.testRestoreUniverseBackup started
[info] Test com.yugabyte.yw.common.TableManagerTest.testCreateS3BackupVerbose started
[info] Test com.yugabyte.yw.common.TableManagerTest.testBulkImportWithIncorrectYBVersion started
[info] Test com.yugabyte.yw.common.TableManagerTest.testRestoreS3Backup started
[info] Test com.yugabyte.yw.common.TableManagerTest.testCreateTLSBackup started
[info] Test com.yugabyte.yw.common.TableManagerTest.testCreateUniverseBackup started
[info] Test com.yugabyte.yw.common.TableManagerTest.testDeleteUniverseBackup started
[info] Test com.yugabyte.yw.common.TableManagerTest.testCreateS3BackupWithSSE started
[info] Test com.yugabyte.yw.common.TableManagerTest.testCreateBackupKubernetes started
[info] Test run finished: 0 failed, 0 ignored, 18 total, 21.681s
[info] Passed: Total 18, Failed 0, Errors 0, Passed 18
[success] Total time: 72 s, completed 22 Jan, 2021 11:25:30 AM

### CODE REVIEW

File: managed/devops/bin/yb_backup.py [hunk 1]
```diff
@@ -1610,6 +1612,30 @@ def delete_bucket_obj(self):
         else:
             self.run_program(del_cmd)
 
+    def find_nfs_storage(self):
```
Comment (Arnav15) [inline managed/devops/bin/yb_backup.py:1615]: As I had mentioned in the our initial discussion, we already allow for running commands in parallel, so there is no point in running these commands sequentially. Check how the `find_data_dirs` function is utilized by the script and do something similar for finding the NFS storage too.

**Review by Arnav15 — CHANGES_REQUESTED** [review]

File: managed/devops/bin/yb_backup.py (hunk 1 above)
Comment (jitendra-12113) [reply managed/devops/bin/yb_backup.py]: Done. Thanks!

File: managed/devops/bin/yb_backup.py [hunk 2]
```diff
@@ -1762,6 +1778,23 @@ def backup_table(self):
         backup location.
         """
 
+        if self.args.storage_type == 'nfs':
```
Comment (Arnav15) [inline managed/devops/bin/yb_backup.py:1781]: We can't do this check only while backing up. We need to do it for restore as well. We have a function called `post_process_arguments` that we can utilize to do this check before running any commands. Further down the line, we can also expand this post processing to eventually verify the validity of the s3, gcs, and azure credentials.

**Review by Arnav15 — CHANGES_REQUESTED** [review]
Sorry for not catching this in my last pass over your diff.

File: managed/devops/bin/yb_backup.py (hunk 2 above)
Comment (jitendra-12113) [reply managed/devops/bin/yb_backup.py]: Yup. Make sense. Thanks for catching this! Updated the PR.

**Review by Arnav15 — APPROVED** [review]
Looks good! Can you also once check the success (if the mount point is present) of the task before merging it in?

**Comment by jitendra-12113** [comment]
> Looks good! Can you also once check the success (if the mount point is present) of the task before merging it in?

Yeah already tested both the scenarios(success and error out). It's working fine. Thanks!

**Pull request merged**